File: kanucontrolClient/src/api/types/simulation/PlanungsSimulation.ts
import { VeranstaltungsInfo } from "./VeranstaltungsInfo";

export interface Beitragsstruktur {
    id: number;
    name: string;
}

export interface PlanungsSimulation {
  status: "IN_BEARBEITUNG" | "EINGEREICHT";
  veranstaltung: VeranstaltungsInfo;
  kikZertifiziert: boolean;
  beitragsstrukturId?: number;
  teilnehmerBeitragUnter21Jahre?: number;
  mitarbeiterBeitrag?: number;
  teilnehmer: number;
  mitarbeiter: number;
  unterkunftPreisProPersonUndNacht?: number;
  verpflegungPreisProPersonUndTag?: number;
  honorare?: number;
  fahrtkosten?: number;
  verbrauchsmaterialProTag?: number;
  sonstigeKostenProTag?: number;
  kultur?: number;
  miete?: number;
}

File: kanucontrolServer/src/main/java/com/kcserver/dto/simulation/PlanungsSimulation.java
package com.kcserver.dto.simulation;

import lombok.*;
import java.math.BigDecimal;

@Builder
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class PlanungsSimulation {

    private VeranstaltungsInfo veranstaltung;

    /**
     * Soll der KiK-Zuschlag berücksichtigt werden?
     */
    private boolean kikZertifiziert;

    /* Teilnehmer */

    private int teilnehmer;
    private int mitarbeiter;

    /* Beiträge */

    private BigDecimal teilnehmerBeitragUnter21Jahre;
    private BigDecimal mitarbeiterBeitrag;

    /* Preise */

    private BigDecimal unterkunftPreisProPersonUndNacht;
    private BigDecimal verpflegungPreisProPersonUndTag;

    /* Kosten */

    private BigDecimal honorare;
    private BigDecimal fahrtkosten;
    private BigDecimal verbrauchsmaterialProTag;
    private BigDecimal kultur;
    private BigDecimal miete;
    private BigDecimal sonstigeKostenProTag;
}
